src/test/ui/consts/rfc-2203-const-array-repeat-exprs/migrate-fail2.rs

```diff
@@ -0,0 +1,15 @@
+// ignore-compare-mode-nll
+// compile-flags: -Z borrowck=migrate
```

Comment:
What is this for?

Reply:
idk, I just copied the test

Reply:
Copied from where?
Also I thought `borrowck=migrate` is on its way out of the compiler, so I am surprised to see new tests for it (Cc @rust-lang/wg-compiler-nll)

Reply:
https://github.com/oli-obk/rust/blob/const_check_cleanup/src/test/ui/consts/rfc-2203-const-array-repeat-exprs/migrate-fail.rs (note the `2` in my version).

Reply:
Okay so the header is the same but the rest of the test is very different.

I still have no clue why this is testing what, and what it has to do with the rest of the PR.

Reply:
So the commit title is "Add test for passing promotion but failing compilation ".

But I don't entirely understand what that is for, doesn't that happen all the time? The `delay_span_bug` is for failinr promotion but passing compilation, i.e., the exact opposite.

Reply:
This test was originally for explicit promotion, so maybe it doesn't make sense anymore